eclair-core/src/test/scala/fr/acinq/eclair/router/GraphSpec.scala

```diff
@@ -261,4 +262,89 @@ class GraphSpec extends AnyFunSuite {
+                 \   /
+                  \ /
+                   E
+    */
+    val edgeAB = makeEdge(1L, a, b, 10001 msat, 0, capacity = 200000 sat)
+    val edgeBC = makeEdge(2L, b, c, 10000 msat, 0, capacity = 200000 sat)
+    val edgeCD = makeEdge(3L, c, d, 10001 msat, 0, capacity = 200000 sat)
+    val edgeDC = makeEdge(4L, d, c, 10 msat, 0, capacity = 200000 sat)
+    val edgeCE = makeEdge(5L, c, e, 10003 msat, 0, capacity = 200000 sat)
+    val edgeDE = makeEdge(6L, d, e, 1 msat, 0, capacity = 200000 sat)
+    val graph = DirectedGraph(Seq(edgeAB, edgeBC, edgeCD, edgeDC, edgeCE, edgeDE))
+
+    val paths = yenKshortestPaths(graph, a, e, 90000000 msat,
+      Set.empty, Set.empty, Set.empty, 2,
+      Left(WeightRatios(1, 0, 0, 0, RelayFees(0 msat, 0))),
+      BlockHeight(714930), _ => true, includeLocalChannelCost = true)
+
+    assert(paths.length == 2)
+    assert(paths.head.path == Seq(edgeAB, edgeBC, edgeCD, edgeDE))
+    assert(paths(1).path == Seq(edgeAB, edgeBC, edgeCE))
+  }
+
+  test("Path C->D->E selected because amount too great for C->E capacity") {
+    /*
+    The channel C -> D -> E is a valid path, but C -> E is not.
+    A --> B --> C <-> D
+                 \   /
+                  \ /
+                   E
+    */
+    val edgeAB = makeEdge(1L, a, b, 10001 msat, 0, capacity = 200000 sat)
+    val edgeBC = makeEdge(2L, b, c, 10000 msat, 0, capacity = 200000 sat)
+    val edgeCD = makeEdge(3L, c, d, 20001 msat, 0, capacity = 200000 sat)
+    val edgeDC = makeEdge(4L, d, c, 10 msat, 0, capacity = 200000 sat)
+    val edgeCE = makeEdge(5L, c, e, 10003 msat, 0, capacity = 10000 sat)
+    val edgeDE = makeEdge(6L, d, e, 1 msat, 0, capacity = 200000 sat)
+    val graph = DirectedGraph(Seq(edgeAB, edgeBC, edgeCD, edgeDC, edgeCE, edgeDE))
+
+    val paths = yenKshortestPaths(graph, a, e, 90000000 msat,
+      Set.empty, Set.empty, Set.empty, 2,
+      Left(WeightRatios(1, 0, 0, 0, RelayFees(0 msat, 0))),
+      BlockHeight(714930), _ => true, includeLocalChannelCost = true)
+
+    // Even though paths to find is 2, we only find 1 because that is all the valid paths that there are.
+    assert(paths.length == 1)
+    assert(paths.head.path == Seq(edgeAB, edgeBC, edgeCD, edgeDE))
+  }
+
+  /**
+   * Find all the shortest paths using the example described in
+   * https://en.wikipedia.org/wiki/Yen%27s_algorithm#:~:text=Yen's%20algorithm
+   */
+  test("all shortest paths are found") {
+    // There will be 3 shortest paths.
+    // Edge capacities are set to be the same so that only feeBase will affect the RichWeight.
+    //  C --> D --> F
+    //   \    ^    /|\
+    //    \   |  /  | \
+    //     \ | /    |  \
+    //       E----->G-->H
+    val edgeCD = makeEdge(1L, c, d, 301 msat, 0, capacity = 100000 sat, minHtlc = 1000 msat)
+    val edgeDF = makeEdge(2L, d, f, 401 msat, 0, capacity = 100000 sat, minHtlc = 1000 msat)
+    val edgeCE = makeEdge(3L, c, e, 201 msat, 0, capacity = 100000 sat, minHtlc = 1000 msat)
```

Comment:
Why 301, 401, 201 and not just 3, 4, 2?

Reply:
No good reason apparently. I changed it to the smaller values and it still worked. For some reason, I was thinking that the behavior would be different if they were too small.